bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/CallOptionsFactory.java

```diff
@@ -39,7 +39,23 @@
    * contain state, specifically start time with an expiration is set; in cases when timeouts are
    * used, implementations should create a new CallOptions each time this method is called.
    */
-  <RequestT> CallOptions create(MethodDescriptor<RequestT, ?> descriptor, RequestT request);
+  <RequestT> CallOptions createForRpc(RequestT request);
+
+  /**
+   * Provide a {@link CallOptions} object to be used in a single RPC attempt, where an RPC operation
+   * may have multiple RPC attempts. {@link CallOptions} can
+   * contain state, specifically start time with an expiration is set; in cases when timeouts are
+   * used, implementations should create a new CallOptions each time this method is called.
+   */
+  <RequestT> CallOptions createForRpcAttempt(RequestT request);
+
+  /**
```

Comment:
Note this javadoc and the corresponding implementation for why we use Optional<Integer> in the other classes above